File: web-client/src/models/group.ts
import { IHomeworkResponse } from "./homework";
import {IUserResponse} from "./user";

export interface IGroupResponse {
  id: string;
  name: string;
  code: string;
  ownerId: string;
  ownerFullName: string;
  students: IUserResponse[];
  teachers: IUserResponse[];
  homeworks: IHomeworkResponse[];
}

export interface IGroupRequest {
  name: string;
  students: string[];
  teachers: string[];
}

File: homework-service/Domain/Group.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace homework_service.Domain
{
    public class Group
    {
        public long _id { get; set; }
        public Guid Id { get; set; }
        public string Name { get; set; }
        public ICollection<User> Students { get; set; }
        public ICollection<User> Teachers { get; set; }
        public string Code { get; set; }
        public IEnumerable<Homework> Homeworks { get; set; }
        public ICollection<Comment> Comments { get; set; }
        public User Owner { get; set; }
    }
}
